fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/rules/RemoveSortRule.java

```diff
@@ -97,6 +99,13 @@ public void onMatch(@Nonnull PlannerRuleCall call) {
             }
         }
 
+        if (!orderingKeysIterator.hasNext()) {
```

Comment:
We cannot mutate a plan in place as there may be other plans ranging over this `RecordQueryPlan`. You need to make a defensive copy here.